File: src/main/java/com/mingshashan/learn/lt/l202/Solution_202.c

bool isHappy(int n){
    while (n != 1 && n != 4) {
        int sum = 0;
        while (n) {
            int mod = n % 10;
            sum += mod * mod;
            n /= 10;
        }
        n = sum;
    }
    return n == 1;
}

File: src/main/java/com/mingshashan/learn/lt/l202/Solution_202.java
package com.mingshashan.learn.lt.l202;

/**
 * Write an algorithm to determine if a number n is happy.
 * <p>
 * A happy number is a number defined by the following process:
 * <p>
 * Starting with any positive integer, replace the number by the sum of the squares of its digits.
 * Repeat the process until the number equals 1 (where it will stay), or it loops endlessly in a cycle which does not include 1.
 * Those numbers for which this process ends in 1 are happy.
 * <p>
 * Return true if n is a happy number, and false if not.
 */
public class Solution_202 {
    public boolean isHappy(int n) {
        while (n != 1 && n != 4) {
            int t = 0;
            while (n > 0) {
                int mod = n % 10;
                t += mod * mod;
                n /= 10;
            }
            n = t;
        }
        return n == 1;
    }
}